pkg/cmd/codespace/common_test.go

```diff
@@ -130,3 +131,171 @@ func Test_codespace_displayName(t *testing.T) {
 		})
 	}
 }
+
+func Test_formatCodespacesForSelect(t *testing.T) {
+	type args struct {
+		codespaces []*api.Codespace
+	}
+	tests := []struct {
+		name                string
+		args                args
+		wantCodespacesNames []string
+		wantCodespacesDirty map[string]bool
+	}{
+		{
+			name: "One codespace: Shows only repo and branch name",
+			args: args{
+				codespaces: []*api.Codespace{
+					{
+						GitStatus: api.CodespaceGitStatus{
+							Ref: "trunk",
+						},
+						Repository: api.Repository{
+							FullName: "cli/cli",
+						},
+						DisplayName: "scuba steve",
+					},
+				},
+			},
+			wantCodespacesNames: []string{
+				"cli/cli: trunk",
+			},
+			wantCodespacesDirty: map[string]bool{},
+		},
+		{
+			name: "Two codespaces on the same repo/branch: Adds the codespace's display name",
+			args: args{
+				codespaces: []*api.Codespace{
+					{
+						GitStatus: api.CodespaceGitStatus{
+							Ref: "trunk",
+						},
+						Repository: api.Repository{
+							FullName: "cli/cli",
+						},
+						DisplayName: "scuba steve",
+					},
+					{
+						GitStatus: api.CodespaceGitStatus{
+							Ref: "trunk",
+						},
+						Repository: api.Repository{
+							FullName: "cli/cli",
+						},
+						DisplayName: "flappy bird",
+					},
+				},
+			},
+			wantCodespacesNames: []string{
+				"cli/cli: scuba steve (trunk)",
+				"cli/cli: flappy bird (trunk)",
+			},
+			wantCodespacesDirty: map[string]bool{},
+		},
+		{
+			name: "Two codespaces on the different branches: Shows only repo and branch name",
+			args: args{
+				codespaces: []*api.Codespace{
+					{
+						GitStatus: api.CodespaceGitStatus{
+							Ref: "trunk",
+						},
+						Repository: api.Repository{
+							FullName: "cli/cli",
+						},
+						DisplayName: "scuba steve",
+					},
+					{
+						GitStatus: api.CodespaceGitStatus{
+							Ref: "feature",
+						},
+						Repository: api.Repository{
+							FullName: "cli/cli",
+						},
+						DisplayName: "flappy bird",
+					},
+				},
+			},
+			wantCodespacesNames: []string{
+				"cli/cli: trunk",
+				"cli/cli: feature",
```

Comment:
❓   How do y'all feel about always showing the display name?

Reply:
That makes sense. I'll follow this up with another PR to suggest a simpler implementation so we can hash out the details there. Thanks!

Reply:
I'd like to see more consistency between `gh cs list` and the picker. I often use `list` to find a codespace that I want to delete and `delete` to delete it, but if I have several with the same repo/branch combo it's impossible to tell which one I want to delete.

Reply:
I agree -- it would be nice to have an output like `gh cs list` with a `> ` cursor on the left to pick one. Implementing that is possible, but might be a bit of work since the library we use is really meant to work with plain strings. As a middle-step I opened a PR to consistently present more information about the codespace: 
 https://github.com/cli/cli/pull/5826